ZenodoExportPlugin.php

```diff
@@ -134,56 +159,107 @@ public function getSettingsFormClassName(): string
+            $republishUpdatedRecord = $this->publishZenodoRecord($object, $zenodoRecId, $recordsApiUrl, $apiKey);
+            if (is_array($republishUpdatedRecord)) {
+                return $republishUpdatedRecord;
+            }
         }
 
-        $responseBody = json_decode($response->getBody());
-        $zenodoRecId = $responseBody->id;
+        // note: can't update files on published records
+        if (!$isUpdate) { // @todo disable file deposit for updates as it will fail if the filename exists already - maybe we should delete all files or the record first in this case
+            $filesDeposit = $this->depositFiles($object, $recordsApiUrl, $apiKey, $zenodoRecId);
+            if (is_array($filesDeposit)) {
+                // File deposit has failed, try to delete the record in Zenodo then return the error
+                $this->deleteRecord($object, $zenodoRecId, $recordsApiUrl, $apiKey);
+                return $filesDeposit;
+            }
+        }
 
-        if (($status = $response->getStatusCode()) != ZENODO_API_DEPOSIT_CREATED) {
-            return [['plugins.importexport.zenodo.register.error.mdsError', $status . ' - ' . $response->getBody()]];
+        // Submit the record to a community
+        $communityId = $this->getCommunityId($context);
+        if ($communityId) {
+            if ($review = $this->createReview($zenodoRecId, $communityId, $recordsApiUrl, $apiKey)) {
+                // @todo depending on the community permissions, this can automatically publish the record - make sure this is clear to the user.
+                $requestId = $this->submitReview($zenodoRecId, $zenodoApiUrl, $apiKey);
+                $autoPublishCommunity = $this->automaticPublishingCommunity($context);
+                if (is_array($requestId)) {
+                    return $requestId;
+                } elseif ($autoPublishCommunity && $requestId) {
+                    // @todo handle errors - if this fails, delete the review request?
+                    $reviewAccepted = $this->acceptReview($requestId, $recordsApiUrl, $apiKey);
+                }
+            } elseif (is_array($review)) {
+                return $review;
+            }
         }
 
-        // if the submission has files
-        $this->depositFiles($object, $zenodoRecId, $url, $apiKey);
+        // @todo consider how to handle this alongside auto publishing in community
+        // (e.g. record may be published already)
+        $autoPublish = $this->automaticPublishing($context);
+        if ($autoPublish) {
+            $published = $this->publishZenodoRecord($object, $zenodoRecId, $recordsApiUrl, $apiKey);
+            if (is_array($published)) {
+                // @todo custom error message? e.g draft was created but publishing failed
+                // in this case there is still a draft - delete it then return the error
+                // @todo can't delete a record if there is an open review to a community
```

Comment:
If we would put this code in front of community insertion, maybe this would not be a problem then?